File: php/DataStructureBenchmark/generate_data.php
<?php

function uuidV4() {
    $data = random_bytes(16);
    $data[6] = chr((ord($data[6]) & 0x0f) | 0x40);
    $data[8] = chr((ord($data[8]) & 0x3f) | 0x80);
    return vsprintf('%s%s-%s-%s-%s-%s%s%s', str_split(bin2hex($data), 4));
}

// 단어 리스트
$animals = ["tiger", "lion", "wolf", "bear", "eagle", "shark", "fox", "whale", "dragon", "panther"];
$natures = ["river", "forest", "ocean", "mountain", "desert", "sky", "storm", "wind", "fire", "ice"];
$concepts = ["energy", "power", "force", "light", "shadow", "speed", "time", "gravity", "signal", "pulse"];
$cosmos = ["galaxy", "star", "planet", "universe", "orbit", "comet", "nebula", "meteor", "cosmos", "void"];

$fp = fopen("data/dataset.txt", "w");

// 헤더
fwrite($fp, "key,keyString,value\n");

$count = 0;
$target = 200000; // 20만

for ($i = 0; $i < $target; $i++) {

    $uuid = uuidV4();

    $a = $animals[array_rand($animals)];
    $b = $natures[array_rand($natures)];
    $c = $concepts[array_rand($concepts)];
    $d = $cosmos[array_rand($cosmos)];

    $keyString = "{$a}-{$b}-{$c}-{$d}";

    $value = "https://example.com/{$a}/{$b}/{$c}/{$d}/" . bin2hex(random_bytes(8));

    fwrite($fp, "{$uuid},{$keyString},{$value}\n");

    // 진행 로그 (optional)
    if ($i % 100000 === 0) {
        echo "Generated: {$i}\n";
    }
}

fclose($fp);

echo "총 {$target}개 생성 완료\n";

File: java/DataStructureBenchmark/src/main/GenerateData.java
package main;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Paths;
import java.util.Random;
import java.util.UUID;

public class GenerateData {
    public static void main(String[] args) throws IOException {
        String[] animals = {"tiger", "lion", "wolf", "bear", "eagle", "shark", "fox", "whale", "dragon", "panther"};
        String[] natures = {"river", "forest", "ocean", "mountain", "desert", "sky", "storm", "wind", "fire", "ice"};
        String[] concepts = {"energy", "power", "force", "light", "shadow", "speed", "time", "gravity", "signal", "pulse"};
        String[] cosmos = {"galaxy", "star", "planet", "universe", "orbit", "comet", "nebula", "meteor", "cosmos", "void"};

        int target = 200000;
        Random rand = new Random();

        // Define the root directory
        String rootDir = "C:/Users/jeonghwanroh/algorithmTest/java/DataStructureBenchmark";
        File dataDir = Paths.get(rootDir, "data").toFile();
        if (!dataDir.exists()) {
            dataDir.mkdirs();
        }

        try (FileWriter fw = new FileWriter(Paths.get(rootDir, "data", "dataset.txt").toFile())) {
            fw.write("key,keyString,value\n");
            for (int i = 0; i < target; i++) {
                String uuid = UUID.randomUUID().toString();
                String a = animals[rand.nextInt(animals.length)];
                String b = natures[rand.nextInt(natures.length)];
                String c = concepts[rand.nextInt(concepts.length)];
                String d = cosmos[rand.nextInt(cosmos.length)];
                String keyString = a + "-" + b + "-" + c + "-" + d;
                String value = "https://example.com/" + a + "/" + b + "/" + c + "/" + d + "/" + Long.toHexString(rand.nextLong());
                fw.write(uuid + "," + keyString + "," + value + "\n");
                if (i % 100_000 == 0) {
                    System.out.println("Generated: " + i);
                }
            }
        }
        System.out.println("총 " + target + "개 생성 완료");
    }
}